File: app/src/main/java/com/benjaminshai/couragers/activities/ActivityWithToolbar.java
package com.benjaminshai.couragers.activities;

import android.content.Intent;
import android.content.pm.PackageManager;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.LinearLayout;

import com.benjaminshai.couragers.R;
import com.benjaminshai.couragers.activities.gallery.GallerySelector;
import com.benjaminshai.couragers.activities.image.upload.UploadActivity;
import com.benjaminshai.couragers.activities.schedule.DaysActivity;

/**
 * Created by bshai on 8/15/15.
 */
public abstract class ActivityWithToolbar extends AppCompatActivity {


    protected void attachToolbar() {
        LinearLayout schedule = (LinearLayout) findViewById(R.id.schedule);

        if (schedule != null) {
            schedule.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Intent i = new Intent(ActivityWithToolbar.this, DaysActivity.class);
                    startActivity(i);
                }
            });
        }

        LinearLayout gallery = (LinearLayout) findViewById(R.id.gallery);

        if(gallery != null) {
            gallery.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Intent i = new Intent(ActivityWithToolbar.this, GallerySelector.class);
                    startActivity(i);
                }
            });
        }

        LinearLayout upload = (LinearLayout) findViewById(R.id.upload);

        if(upload != null) {
            if (getPackageManager().hasSystemFeature(PackageManager.FEATURE_CAMERA)) {
                upload.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        Intent i = new Intent(ActivityWithToolbar.this, UploadActivity.class);
                        startActivity(i);
                    }
                });
            } else {
                upload.setVisibility(View.GONE);
            }
        }
    }
}

File: app/src/main/java/com/benjaminshai/couragers/activities/ActivityWithToolbar.kt
package com.benjaminshai.couragers.activities

import android.content.Intent
import android.content.pm.PackageManager
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.LinearLayout

import com.benjaminshai.couragers.Constants
import com.benjaminshai.couragers.R
import com.benjaminshai.couragers.activities.gallery.GalleriesActivity
import com.benjaminshai.couragers.activities.image.upload.UploadActivity
import com.benjaminshai.couragers.activities.schedule.DaysActivity

/**
 * Created by bshai on 8/15/15.
 */
abstract class ActivityWithToolbar : AppCompatActivity() {


    protected fun attachToolbar() {
        val schedule = findViewById<View>(R.id.schedule) as LinearLayout

        schedule?.setOnClickListener {
            val i = Intent(this@ActivityWithToolbar, DaysActivity::class.java)
            startActivity(i)
        }

        val gallery = findViewById<View>(R.id.gallery) as LinearLayout

        gallery?.setOnClickListener {
            val i = Intent(this@ActivityWithToolbar, GalleriesActivity::class.java)
            i.putExtra("collectionId", Constants.KIDS_COLLECTION_ID)
            startActivity(i)
        }

        val upload = findViewById<View>(R.id.upload) as LinearLayout

        if (upload != null) {
            if (packageManager.hasSystemFeature(PackageManager.FEATURE_CAMERA)) {
                upload.setOnClickListener {
                    val i = Intent(this@ActivityWithToolbar, UploadActivity::class.java)
                    startActivity(i)
                }
            } else {
                upload.visibility = View.GONE
            }
        }
    }
}
